csharp/src/Ice/Endpoint.cs

```diff
@@ -40,17 +41,19 @@ public abstract class Endpoint : IEquatable<Endpoint>
 
         /// <summary>Gets an option of the endpoint.</summary>
         /// <param name="option">The name of the option to retrieve.</param>
-        /// <value>The value of this option, or null if this option is not set.</value>
+        /// <value>The value of this option, or null if this option is unknown, not set or set to its default value.
+        /// </value>
         public virtual string? this[string option]
         {
             get
             {
                 if (Protocol == Protocol.Ice1)
                 {
-                    // Convert ice1 API into options
                     return option switch
                     {
                         "compress" => HasCompressionFlag ? "true" : null,
+                        "host" => Host.Length > 0 ? Host : null,
+                        "port" => Port != DefaultPort ? Port.ToString(CultureInfo.InvariantCulture) : null,
                         "timeout" => Timeout != DefaultTimeout ?
                             Timeout.TotalMilliseconds.ToString(CultureInfo.InvariantCulture) : null,
```

Comment:
If we are not going to use the endpoint compression flag and timeout anymore, we could get rid of them from the endpoint base class and only deal with them in the TcpEndpoint class like we deal with the resource data member in the WsEndpoint.

Reply:
Yes, I hope we can clean this up in a follow-up PR.